### DISCUSSION TITLE
How to override existing module view?

### DISCUSSION THREAD

**Discussion by lampersky** [OrchardCMS/OrchardCore#8652] [Q&A] [answered]
I'm trying to override few views (cshtml):
- BagPart.Edit.cshtml (from **OrchardCore.Flows** module),
- MediaField.Edit.cshtml (from **OrchardCore.Media** module)

by providing own implementation within custom theme.

Is this even possible? Can I override them same way as it is described here https://docs.orchardcore.net/en/dev/docs/reference/modules/Templates/#overriding-login-view for login view?

Here is how my theme structure looks like:
![image](https://user-images.githubusercontent.com/6403130/108716848-e5cec700-751c-11eb-9b25-6931510d4680.png)

Manifest.cs has dependencies defined for both modules.

**Comment by deanmarcussen** [comment, marked as answer]
You either need to do this from an AdminTheme or from a Module. Either works, though for modules, you need to define your dependencies, so that the new views override the old ones.

But from a normal front end theme, it will not be the selected theme, so the views will not be discovered when viewing the admin.

**Reply by lampersky** [reply]
> But from a normal front end theme, it will not be the selected theme, so the views will not be discovered when viewing the admin.

makes sense :) Thank you @deanmarcussen 

I will try from module then.

In the meantime I was trying to override views from application level, and here I achieve partial success:
- BagPart.Edit.cshtml **is** overridden,
- MediaField.Edit.cshtml **isn't** overridden.

Here is my app:
![image](https://user-images.githubusercontent.com/6403130/108728106-8a570600-7529-11eb-83ae-4d1e77bf2531.png)

What's the difference between those two views, one can be overridden and the other not.
Or maybe I missed something.

**Reply by Skrypt** [reply]
Views that are standing in a /Views/Shared folder can be overridden from that same folder directly. Though, if a View is standing directly in the /Views folder you need to override it by using a subfolder in the Views folder corresponding to the module Area name. Each modules is a different Area so for the Media module it would be /Views/OrchardCore.Medias/MediaField.Edit.cshtml "I think", without trying.

One trick I use to see where it is harvesting these is that I simply rename the View template as .cshtml.bak and then go to the page where they are called and I get that error message where it shows where it's trying to find that view.

**Reply by lampersky** [reply]
> One trick I use to see where it is harvesting these is that I simply rename the View template as .cshtml.bak and then go to the page where they are called and I get that error message where it shows where it's trying to find that view.

I'm using OC from nuget packages. But thank you @Skrypt for the trick

also I think I was trying with area name too (with no luck), but will check and make sure again

**Reply by lampersky** [reply]
I've just tried with Module, and everything works, both views got overridden.
Thanks guys!
